content_server/index/local.py

```diff
@@ -0,0 +1,242 @@
+            f"min_version={self.min_version!r}, "
+            f"max_version={self.max_version!r}, "
+            f"tags={self.tags!r})"
+        )
+
+
+class Index:
+    def _read_content_entry_version(self, content_type, unique_id, data, md5sum_mapping, get_next_content_id):
+        unique_id = bytes.fromhex(unique_id)
+
+        md5sum_partial = bytes.fromhex(data["md5sum-partial"])
+        md5sum = md5sum_mapping[content_type][unique_id][md5sum_partial]
+
+        content_id = get_next_content_id(content_type, unique_id, md5sum)
+
+        dependencies = []
+        for dependency in data.get("dependencies", []):
+            dep_content_type = get_content_type_from_name(dependency["content-type"])
+            dep_unique_id = bytes.fromhex(dependency["unique-id"])
+
+            dep_md5sum_partial = bytes.fromhex(dependency["md5sum-partial"])
+            dep_md5sum = md5sum_mapping[dep_content_type][dep_unique_id][dep_md5sum_partial]
+
+            dependencies.append((dep_content_type, dep_unique_id, dep_md5sum))
+
+        min_version = None
+        max_version = None
+        for com in data.get("compatibility", {}):
+            if com["name"] != "master":
+                continue
+
+            for conditions in com["conditions"]:
+                if conditions.startswith(">="):
+                    min_version = [int(p) for p in conditions[2:].split(".")]
+                elif conditions.startswith("<"):
+                    max_version = [int(p) for p in conditions[1:].split(".")]
+                else:
+                    raise Exception("Invalid compatibility flag", com)
+
+        # Validate the object to make sure all fields are within set limits.
+        ContentEntryTest().load(
+            {
+                "content-type": content_type,
+                "content-id": content_id,
+                "filesize": data["filesize"],
+                "name": data["name"],
+                "version": data["version"],
+                "url": data["url"],
+                "description": data["description"],
+                "unique-id": unique_id,
+                "md5sum": md5sum,
+                "min-version": min_version,
+                "max-version": max_version,
+                "tags": data.get("tags", []),
+                "raw-dependencies": dependencies,
+            }
+        )
+
+        # Calculate if this entry wouldn't exceed the OpenTTD packet size if
+        # we would transmit this over the wire.
+        size = 1 + 4 + 4  # content-type, content-id, filesize
+        size += len(data["name"])
+        size += len(data["version"])
+        size += len(data["url"])
+        size += len(data["description"])
+        size += len(unique_id)
+        size += len(md5sum)
+        size += len(dependencies) * 4
+        size += 1
+        for tag in data.get("tags", []):
+            size += len(tag)
+
+        if size > 1400:
+            raise Exception("Entry would exceed OpenTTD packet size.")
```

Comment:
Would be nice to have a specific exception type rather than just a generic `Exception` so you can catch the specific errors